packages/autocomplete-js/src/debounce.ts

```diff
@@ -0,0 +1,8 @@
+export function debounce(fn: Function, time: number) {
+  let timerId: number | undefined = undefined;
+
+  return function () {
+    clearTimeout(timerId);
+    timerId = (setTimeout(fn, time) as unknown) as number;
+  };
+}
```

Comment:
```suggestion
type TimerId = ReturnType<typeof setTimeout>;

export function debounce(fn: Function, time: number) {
  let timerId: TimerId | undefined = undefined;

  return function () {
    clearTimeout(timerId);
    timerId = setTimeout(fn, time);
  };
}
```

this way it works both in node (`NodeJS.Timer`) & browser (number)

Reply:
actually you might need to guard for `timerId` in node:

```ts
    if (timerId) {
      clearTimeout(timerId);
    }
```

Reply:
Thanks!